### PR TITLE
Fix URL that produce Project API

### PR DESCRIPTION
**Pull request by vparfonov**
### What does this PR do?
Fixes URL provided by Project API. Mainly this is needed for Hosted version (Codenvy package). 
In multi node installation we have some additional network configuration. It produces additional path param in URL to wsagent which describe node domain and port where wsagent is launched. So to resolve URL correctly we removed rewrite rules from Tomcat and changed mapping for REST Servlet and WebSocket Endpoint. 

Related issue https://github.com/codenvy/codenvy/issues/1076

Signed-off-by: Vitalii Parfonov <vparfonov@codenvy.com>

### CODE REVIEW

**Comment by vparfonov** [comment]
@skabashnyuk

File: assembly/onpremises-ide-packaging-war-ext-server/src/main/java/com/codenvy/ext/java/server/CodenvyWebSocketEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.eclipse.che.api.core.websocket.impl.GuiceInjectorEndpointConfigurator;
+import org.eclipse.che.api.core.websocket.impl.PendingMessagesReSender;
+import org.eclipse.che.api.core.websocket.impl.WebSocketSessionRegistry;
+import org.eclipse.che.api.core.websocket.impl.WebSocketTransmissionDispatcher;
+
+import javax.inject.Inject;
+import javax.websocket.server.ServerEndpoint;
+
+/**
+ * @author Vitalii Parfonov
```
Comment (skabashnyuk) [inline assembly/onpremises-ide-packaging-war-ext-server/src/main/java/com/codenvy/ext/java/server/CodenvyWebSocketEndpoint.java]: What is it.

File: wsmaster/codenvy-hosted-machine-authentication-ide/src/main/java/com/codenvy/machine/authentication/ide/MachineAsyncRequestFactory.java [hunk 2]
```diff
@@ -112,4 +126,28 @@ public String apply(MachineTokenDto tokenDto) throws FunctionException {
...
     }
+
+    private boolean isWsAgentRequest(String url) {
+        if (isNullOrEmpty(wsAgentBaseUrl)) {
+            return false; //ws-agent not started
+        }
+        return url.startsWith(wsAgentBaseUrl);
+    }
+
+    @Override
```
Comment (skabashnyuk) [inline wsmaster/codenvy-hosted-machine-authentication-ide/src/main/java/com/codenvy/machine/authentication/ide/MachineAsyncRequestFactory.java]: what are you doing here?

File: wsmaster/codenvy-hosted-machine-authentication-ide/src/main/java/com/codenvy/machine/authentication/ide/MachineAsyncRequestFactory.java [hunk 3]
```diff
@@ -112,4 +126,28 @@ public String apply(MachineTokenDto tokenDto) throws FunctionException {
     public void onWorkspaceStopped(WorkspaceStoppedEvent event) {
         machineToken = null;
     }
+
```
Comment (skabashnyuk) [inline wsmaster/codenvy-hosted-machine-authentication-ide/src/main/java/com/codenvy/machine/authentication/ide/MachineAsyncRequestFactory.java]: what is it?

**Pull request merged**